src/Preprocessor.zig

```diff
@@ -1232,6 +1232,43 @@ fn expandFuncMacro(
                     some.expansionSlice(),
                 ) else try pp.pragmaOperator(string.?, loc);
             },
+            .comma => {
+                if (tok_i + 2 < func_macro.tokens.len) {
+                    const maybe_hashhash = func_macro.tokens[tok_i + 1];
+                    const maybe_va_args = func_macro.tokens[tok_i + 2];
+                    if (maybe_hashhash.id == .hash_hash and maybe_va_args.id == .keyword_va_args) {
+                        // GNU extension: `, ##__VA_ARGS__` deletes the comma if __VA_ARGS__ is empty
+                        tok_i += 2;
+                        if (func_macro.params.len == expanded_args.items.len) {
+                            // Empty __VA_ARGS__, drop the comma
+                            try pp.err(maybe_hashhash, .comma_deletion_va_args);
+                        } else if (func_macro.params.len == 0 and expanded_args.items.len == 1 and expanded_args.items[0].len == 0) {
+                            // Ambiguous whether this is "empty __VA_ARGS__" or "__VA_ARGS__ omitted"
+                            if (pp.comp.langopts.standard.isGNU()) {
+                                // GNU standard, drop the comma
+                                try pp.err(maybe_hashhash, .comma_deletion_va_args);
+                            } else {
+                                // C standard, retain the comma
+                                try buf.append(tokFromRaw(raw));
+                            }
+                        } else {
+                            try buf.append(tokFromRaw(raw));
+                            if (expanded_variable_arguments.items.len > 0) {
+                                try pp.err(maybe_hashhash, .comma_deletion_va_args);
```

Comment:
Clang gives this error even when that condition is false:
```c
#define foo(a,...) a,##__VA_ARGS__
#define bar

foo(1,bar)
```

Reply:
I added a condition to check for this (`variable_arguments.items.len == func_macro.params.len`) but I'm not totally sure it's the correct check or that it exactly matches what GCC does. I also updated it to ignore `.macro_ws` between `##` and `__VA_ARGS__`